infra/Pos/KnownPeers.hs

```diff
@@ -13,5 +14,7 @@ class MonadKnownPeers m where
   updateKnownPeers :: (Peers NodeId -> Peers NodeId) -> m ()
   addKnownPeers    :: Peers NodeId -> m ()
   removeKnownPeer  :: NodeId -> m ()
-  -- | Good for debugging purposes.
-  formatKnownPeers :: (forall a . Format r a -> a) -> m r
+
+-- | For debugging: return formatted list of peers, if available
+class MonadFormatPeers m where
+  formatKnownPeers :: (forall a . Format r a -> a) -> m (Maybe r)
```

Comment:
I think it's because we need it in some places where we don't have an outbound queue.